src/createServer.js

```diff
@@ -1,3 +1,53 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+function createServer() {
+  return http.createServer((req, res) => {
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const wordToConvert = normalizedURL.pathname.slice(1);
+    const typeConvent = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+    const messages = {
+      badCase: 'This case is not supported. '
+      + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      noText: 'Text to convert is required. '
+      + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      noCase: '"toCase" query param is required. '
+      + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+    };
+
+    const errors = [];
+    let body = {};
+
+    const requestCase = normalizedURL.searchParams.get('toCase');
+
+    if (!normalizedURL.search) {
+      errors.push({ message: messages.noCase });
+    }
+
+    if (!typeConvent.includes(requestCase) && normalizedURL.search) {
+      errors.push({ message: messages.badCase });
+    } else if (typeConvent.includes(requestCase)) {
+      body = convertToCase(wordToConvert, requestCase);
+    }
+
+    if (!wordToConvert) {
+      errors.push({ message: messages.noText });
+    }
+
+    body.targetCase = requestCase;
+    body.originalText = wordToConvert;
```

Comment:
you can create here body. It will be easier to read your code
```suggestion
    }

    if (!wordToConvert) {
      errors.push({ message: messages.noText });
    }

    const {...} = convertToCase(wordToConvert, requestCase);
    body = {
      targetCase: requestCase,
      ...
    }
```